CODING_GUIDELINES.md

```diff
@@ -25,6 +25,13 @@ We deviate from the [Google Python Style Guide][google-style-guide] only in the
 - We use [`black`][black] and [`isort`][isort] for source code and imports formatting, which may work differently than indicated by the guidelines in section [_3. Python Style Rules_](https://google.github.io/styleguide/pyguide.html#3-python-style-rules). For example, maximum line length is set to 100 instead of 79 (although docstring lines should still be limited to 79).
 - According to subsection [_2.19 Power Features_](https://google.github.io/styleguide/pyguide.html#219-power-features), direct use of _power features_ (e.g. custom metaclasses, import hacks, reflection) should be avoided, but standard library classes that internally use these power features are accepted. Following the same spirit, we allow the use of power features in infrastructure code with similar functionality and scope as the Python standard library.
 - According to subsection [_3.19.12 Imports For Typing_](https://google.github.io/styleguide/pyguide.html#31912-imports-for-typing), symbols from `typing` and `collections.abc` modules used in type annotations _"can be imported directly to keep common annotations concise and match standard typing practices"_. Following the same spirit, we allow symbols to be imported directly from third-party or internal modules when they only contain a collection of frequently used typying definitions.
+- Imports [_2.2 Imports_](https://google.github.io/styleguide/pyguide.html#22-imports):
+  - Importing individual classes and functions is allowed if they are to be used in DSL code and the DSL does not allow them to be qualified.
+  - Importing individual classes and functions is allowed in order to explicitly re-export them (via `__all__`). Each module which is allowed to re-export code objects has to be documented here with a reason.
+    - `gt4py.next` reexports FieldView classes and helpers in order to create a `numpy` like user experience.
+  - `from path.to.submodule import *` is **only** allowed in order to re-export the contents of a submodule if:
+    - the submodule in question defines `__all__` and
+    - the submodule in question exports many public API objects (example: `gt4py.next.ffront.fbuiltins`).
```

Comment:
Again this is very specific to our current codebase, so I would move these points to a module docstring. The point about defining `__all__` in modules which will be re-exported it's already more or less already documented in the point 5 of the `Module structure` section, so I think it could be enough with expanding a bit that point, e.g.
`5. Definition of exported symbols (optional in general, but required for re-exporting symbols from other modules):`